src/System.Text.Json/src/System/Text/Json/Node/JsonObject.cs

```diff
@@ -526,41 +523,54 @@ public bool TryGetJsonArrayPropertyValue(string propertyName, out JsonArray json
             return false;
         }
 
+        /// <summary>
+        ///   Returns the JSON array value of a property with the specified name.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to return.</param>
+        /// <param name="jsonArray">JSON array value of the property with specified name.</param>
+        /// <param name="stringComparison">The culture and case to be used when comparing string value.</param>
+        /// <returns>
+        ///  <see langword="true"/> if JSON array property with specified name was found;
+        ///  otherwise, <see langword="false"/>
+        /// </returns>
+        public bool TryGetJsonArrayPropertyValue(string propertyName, out JsonArray jsonArray, StringComparison stringComparison)
+        {
+            if (TryGetPropertyValue(propertyName, out JsonNode jsonNode, stringComparison))
+            {
+                jsonArray = jsonNode as JsonArray;
+                return jsonArray != null;
+            }
+
+            jsonArray = null;
+            return false;
+        }
+
         /// <summary>
         ///   A collection containing the property names of JSON object.
         /// </summary>
-        public ICollection<string> PropertyNames => _dictionary.Keys;
+        public ICollection<string> GetPropertyNames() => _dictionary.Keys;
```

Comment:
Hmmm, I wonder if `IEnumerable<string>` or `IReadOnlyCollection<string>` would be better here. We don't want people to think they can modify the `JsonObject` via this collection.

Reply:
IDictionary properties `Keys` and `Values` have this signature, but as we are not implementing the interface and decided to make some changes, probably it would be valuable to have this change as well.